nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/ListAzureDataLakeStorage.java

```diff
@@ -129,6 +131,16 @@
             .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
             .build();
 
+    public static final PropertyDescriptor TEMP_FILE_PREFIX = new PropertyDescriptor.Builder()
+            .name("azure-temp-file-prefix")
+            .displayName("Temp File Prefix for Azure")
```

Comment:
The concept seems straightforward, but given the behavior, would it make sense to rename this to something more general?  Without reading the description, at first it sounds like the property will be used to create temporary files as opposed to filtering them.  What do you think about renaming this property and associated variables to something like `Filter File Prefix`?  Taking the concept one step further, would it make sense to change this to a Regular Expression pattern instead of a prefix?

In either case the `for Azure` portion of the property display name seems unnecessary since this component is already specific to Azure.

Reply:
Thanks for pointing out the other filter properties @jfrazee.  I agree with you, it sounds like it would be cleaner to set a default value for the File Filter property to exclude temporary files as opposed to introducing a new property.

Reply:
@exceptionfactory and @jfrazee Thank you very much for your help. I've implemented the requested changes however during the unit tests it has brought up another perspective I would like to share. 
Combining the file filter with the temp prefix would mean we would like to have a regex which would like to include only the specified files and exclude some specified files, I think this is very confusing for the user and may not be a best practice.
As we need to provide the option for prefix override it means that sometimes separate Regex expressions must be concatenated. I have no knowledge regarding our Users' Regex experience level however combining two separate, conflicting scoped (for example a negative lookahead and a wildcard word match) Regex Expression can be challenging even for Developers.
My last concern is what if the Processor already has a path filter set. It would override the new default filter and temp files started appear in the listing which may impact the existing flows.
So for the best User experience I would like to recommend to keep the prefix property which serves as the default value and the override possibility however change it to accept a Regex expression. This way our Users have the control to change only the filter they have to in a straightforward and convenient way.

Reply:
Thanks for following up and noting those concerns @timeabarna, I can see the challenge of defining complex regular expressions to filter temporary files and other files.  Looking at other processors, `ListSFTP` includes a Boolean property for `Ignored Dotted Files` as well as properties for filtering files and paths using regular expressions.  It seems like the question is whether it is necessary to support filtering both temporary files and other files.  On the one hand, having one filter property avoids confusion from potentially conflicting patterns, but on the other hand, creating the right expression could be challenging.

@jfrazee may have a better understanding of the runtime environment, but do temporary files always follow a standard pattern?  If so, perhaps having a Boolean property for filtering temporary files would be easier to understand and provide the desired behavior.

Reply:
> @jfrazee may have a better understanding of the runtime environment, but do temporary files always follow a standard pattern? If so, perhaps having a Boolean property for filtering temporary files would be easier to understand and provide the desired behavior.

I may not have been thinking clearly because I was thinking of files like _SUCCESS and _temporary which are used more like dotted files in HDFS. I think it'll be unlikely to see those though unless someone just did a bulk transfer from HDFS to ADLS so it's probably a non-concern unless someone says they need it.

Reply:
@exceptionfactory @jfrazee So if I understand you correctly, the prefix property should be removed from both put and list processors, having a built-in prefix instead. List processor should have a boolean property whether temp files should be included in the listing. Is this correct?

Reply:
@exceptionfactory @jfrazee Can you please confirm my understanding, so I can start modifying the code accordingly? Thanks in advance

Reply:
@timeabarna Changing the approach to use a Boolean property with hard-coded temporary prefixes seems to be a better way to go if @jfrazee agrees that temporary files always follow the same naming convention.

Reply:
@exceptionfactory @jfrazee I've refactored the code using built-in prefix and added boolean property, thanks in advance for checking this PR again